### ISSUE TITLE
error "stack level too deep" after updating from 10.0.1 to 10.0.5

### ISSUE THREAD

**Issue by MathieuDerelle** [airbrake/airbrake#1104]
The error was not present when using 10.0.1.

there is a conflict with newrelic on the instrumentation of http calls

the last line of the backtrace is my method that calls Net::Http
then your code and newrelic's code are looping on each others

```
...
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:27:in `block (2 levels) in request_with_newrelic_trace'
newrelic_rpm (3.16.0.318) lib/new_relic/agent.rb:428:in `disable_all_tracing'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:26:in `block in request_with_newrelic_trace'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/cross_app_tracing.rb:41:in `tl_trace_http_request'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:23:in `request_with_newrelic_trace'
airbrake (10.0.5) lib/airbrake/rails/net_http.rb:11:in `block in request'
airbrake (10.0.5) lib/airbrake/rack.rb:26:in `block in capture_timing'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:29:in `span'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `block in span'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `tap'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `span'
airbrake (10.0.5) lib/airbrake/rack.rb:25:in `capture_timing'
airbrake (10.0.5) lib/airbrake/rails/net_http.rb:10:in `request'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:27:in `block (2 levels) in request_with_newrelic_trace'
newrelic_rpm (3.16.0.318) lib/new_relic/agent.rb:428:in `disable_all_tracing'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:26:in `block in request_with_newrelic_trace'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/cross_app_tracing.rb:41:in `tl_trace_http_request'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:23:in `request_with_newrelic_trace'
airbrake (10.0.5) lib/airbrake/rails/net_http.rb:11:in `block in request'
airbrake (10.0.5) lib/airbrake/rack.rb:26:in `block in capture_timing'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:29:in `span'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `block in span'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `tap'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `span'
airbrake (10.0.5) lib/airbrake/rack.rb:25:in `capture_timing'
airbrake (10.0.5) lib/airbrake/rails/net_http.rb:10:in `request'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:27:in `block (2 levels) in request_with_newrelic_trace'
newrelic_rpm (3.16.0.318) lib/new_relic/agent.rb:428:in `disable_all_tracing'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:26:in `block in request_with_newrelic_trace'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/cross_app_tracing.rb:48:in `tl_trace_http_request'
newrelic_rpm (3.16.0.318) lib/new_relic/agent/instrumentation/net.rb:23:in `request_with_newrelic_trace'
airbrake (10.0.5) lib/airbrake/rails/net_http.rb:11:in `block in request'
airbrake (10.0.5) lib/airbrake/rack.rb:26:in `block in capture_timing'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:29:in `span'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `block in span'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `tap'
airbrake-ruby (4.15.0) lib/airbrake-ruby/timed_trace.rb:18:in `span'
airbrake (10.0.5) lib/airbrake/rack.rb:25:in `capture_timing'
airbrake (10.0.5) lib/airbrake/rails/net_http.rb:10:in `request'
lib/receipt_processing/interface.rb:107:in `call_api'
...
```

gemlock before : 
```
    airbrake (10.0.1)
      airbrake-ruby (~> 4.13)
    airbrake-ruby (4.13.3)
```
after : 
```
    airbrake (10.0.5)
      airbrake-ruby (~> 4.13)
    airbrake-ruby (4.15.0)
```

**Comment by MathieuDerelle** [comment]
this issue is also present on 10.0.4 (tried it for #1072)

**Comment by kyrylo** [comment]
Hi, thanks for the report. The `new_relic` gem needs to stop monkey-patching using class reopening and "alias_method_chain": https://github.com/newrelic/rpm/blob/94542740155b32f0e6298ecf8932cec41e8816da/lib/new_relic/agent/instrumentation/net.rb#L19-L20

We had a similar problem where Airbrake was doing `class_eval` to monkey-patch `Net::HTTP`. This resulted in a conflict with other gems. New Relic needs to use `prepend` instead. We used to make the same mistake but it got fixed in https://github.com/airbrake/airbrake/pull/1078/files, which is why you stumbled upon this error.

A simple fix for you would be reordering `new_relic` and `airbrake` in your Gemfile.

**Comment by MathieuDerelle** [comment]
thanks, reordering `require` statements between `new_relic` and `airbrake` (`new_relic` first) fixed it for me.

Also submitted an issue to New Relic

**Comment by bcaplan** [comment]
I've been following this issue since we ran into the same problem. New Relic just released version 7.2.0 which seems to address this incompatibility:

https://github.com/newrelic/newrelic-ruby-agent/blob/583dbe842162f632274d03cee455ff5c1d440e48/CHANGELOG.md#v720
